addons/account/static/src/components/char_with_placeholder_field/char_with_placeholder_field.js

```diff
@@ -0,0 +1,50 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { CharField, charField } from "@web/views/fields/char/char_field";
+import { formatChar } from "@web/views/fields/formatters";
+
+export class CharWithPlaceholderField extends CharField {
+    static template = "account.CharWithPlaceholderField";
+    static props = {
+        ...CharField.props,
+        placeholderField: { type: String },
+    };
+
+    /** Override **/
+    get formattedValue() {
+        const value =
+            this.props.record.data[this.props.name] ||
+            this.props.record.data[this.props.placeholderField];
+        return formatChar(value, { isPassword: this.props.isPassword });
+    }
+
+    get placeholder() {
+        return this.props.record.data[this.props.placeholderField] || this.props.placeholder;
+    }
+}
+
+// Since the `onchangeOnKeydownMixin` modifies `charField.extractProps`, we isolate a reference to the
+// function here to ensure that `CharField` and `charField.extractProps` are in sync.
+const charExtractProps = charField.extractProps;
```

Comment:
The code from `mail` is supposed to run before this, right? So the patch should already be done anyway?
I don't understand what this is changing.

Reply:
You're right - normally it should.
I did this because when running the test, it seems that the code from `mail` runs afterwards, for some reason, and in that case the `charField.extractProps` tries to extract props that aren't defined in our `CharWithPlaceholderField.props`. But I agree that's not normal.

I'll investigate that and see what, if anything, is wrong with the test runner.

Reply:
Hello @antoine162 

Nice explanation. Indeed, it seems that there's an issue with hoot's module sets. We don't know the dependencies defined in the manifests, but I think we should keep the order the modules appear in the bundle. I assume that's what's happening outside of the test environment. cc @jum-odoo 

For now, I would add the dummy dependency to `@mail/js/onchange_on_keydown` s.t. the order is determined and you don't have a different behavior in/outside tests (with a comment explaining why we do this).

Also, I don't think that we should have this patch in mail... Ideally it should be a specific widget, not an option. But they couldn't because this needed to be composed with other features (widget `text_emojis`), I think. I would move the support of the option directly in web (or challenge the feature and remove it...).

Reply:
Thanks so much :smile: